auditlog/iter.go

```diff
@@ -0,0 +1,67 @@
+package auditlog
+
+import (
+	"reflect"
+)
+
+// Query is the function used to get a page listing.
+type Query func(ListRequestParams) ([]interface{}, ListMeta, error)
+
+// EventItr represents and iterative event
+type Iter struct {
+	values  []interface{}
+	current interface{}
+	err     error
+	params  ListRequestParams
+	meta    ListMeta
+	query   Query
+}
+
+// Next returns the next event in the set
```

Comment:
If the Iter interface is for any resource maybe it should be:
```suggestion
// Next returns the next value in the set
```